Assets/Tests/Common/CodeCoverageHistoryCleaner.cs

```diff
@@ -0,0 +1,170 @@
+
+                // Get all CoverageHistory.xml files in the directory
+                string[] coverageFiles = Directory.GetFiles(CODE_COVERAGE_HISTORIC_DATA_DIRECTORY, "*CoverageHistory.xml", SearchOption.AllDirectories);
+
+                Dictionary<DateTime, XElement> lastEntryPerDay = new();
+
+                // For each file
+                foreach (string file in coverageFiles)
+                {
+                    XDocument doc = XDocument.Load(file);
+
+                    var entries = doc.Descendants("coverage")
+                        .Select(element => new
+                        {
+                            DateTime.ParseExact(element.Attribute("date")?.Value, CODE_COVERAGE_DATETIME_FORMAT, null).Date,
+                            Element = element
+                        });
+
+                    // For each entry in the file
+                    foreach (var entry in entries)
+                    {
+                        // If there already is an entry for the indicated date.
+                        if (lastEntryPerDay.ContainsKey(entry.Date))
+                        {
+                            // Replace the entry if the current one happened
+                            // later in the day.
+                            var existingEntryTime = DateTime.ParseExact(lastEntryPerDay[entry.Date].Attribute("date")?.Value, CODE_COVERAGE_DATETIME_FORMAT, null);
+                            var newEntryTime = DateTime.ParseExact(entry.Element.Attribute("date")?.Value, CODE_COVERAGE_DATETIME_FORMAT, null);
+
+                            if (newEntryTime > existingEntryTime)
+                            {
+                                lastEntryPerDay[entry.Date] = entry.Element;
+                            }
+                        }
+                        else
+                        {
+                            // If there isn't an entry for the indicated date,
+                            // then add the current entry for the date.
+                            lastEntryPerDay[entry.Date] = entry.Element;
+                        }
+                    }
+                }
+
+                // Create a new XML document for the consolidated history
+                XElement root = new("CoverageHistory", lastEntryPerDay.Values);
+
+                // Save the contents of the consolidated history. Omit any 
+                // duplicate namespaces and disable formatting, since this file
+                // should only ever be read by the code coverage utility itself.
+                root.Save(outputFilePath, SaveOptions.DisableFormatting | SaveOptions.OmitDuplicateNamespaces);
+
+                Debug.Log($"Consolidated CoverageHistory saved to {outputFilePath}");
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError($"CoverageHistoryCleaner failed: {ex.Message}");
```

Comment:
Consider Debug.LogException

Reply:
Good suggestion! I just changed it to that.